### PR TITLE
feat: implemented `RepeatAction` wrapper

### PR DESCRIPTION
**Pull request by Markus28**
# Description

Implements feature request #652. This still needs a unit test.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Kallinteris-Andreas** [comment]
@pseudorndthoughts Do we actually want this feature? In the original issue It was proposed for the use in MuJoCo environments, but MuJoCo already has The `frame_skip` argument which achieves effectively the same result.

File: gymnasium/wrappers/stateful_action.py [hunk 1]
```diff
@@ -80,3 +84,71 @@ def action(self, action: ActType) -> ActType:
...
+        """Initialize RepeatAction wrapper.
+
+        Args:
+            env (Env): the wrapped environment
+            num_repeats (int): the maximum number of times to repeat the action
+        """
+        if num_repeats <= 1:
+            raise ValueError(
+                f"Number of action repeats should be greater than 1, but got {num_repeats}"
+            )
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/wrappers/stateful_action.py:128]: Add check that `num_repeats` is an integer (supporting numpy integers)

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Thanks for the PR @Markus28 
I just realised that this is very similar to `MaxAndSkipObservation` (https://github.com/Farama-Foundation/Gymnasium/blob/fb5112fbcafdc65dcd6c65fcbf2f7920d84c6aa1/gymnasium/wrappers/stateful_observation.py#L536)

@Kallinteris-Andreas It is a simple wrapper that might be helpful for users doing Atari style environment details

If we add this, we will need to add tests and to the website wrapper docs

**Comment by Kallinteris-Andreas** [comment]
This PR has been inactive for more than 6 months, therefore I am closing it

@Markus28 if you would like to resume, feel free to merge main and continue the work

**Pull request closed without merging**